static/report/js/compare.js

```diff
@@ -0,0 +1,639 @@
+    query.end = end;
+  }
+
+  function param_string (begin, end) {
+    return [
+      "find[dateString][$gte]=" + Date.create(begin).format(DATE_FMT)
+    , "find[dateString][$lte]=" + Date.create(end).format(DATE_FMT)
+    ].join('&')
+  }
+
+  function iter (prev, current, index, tail) {
+  }
+
+  function iter_query (range) {
+    payload = [ ];
+    range.forEach(function (day, i) {
+      var start = day;
+      var end = d3.time.day.offset(start, 1);
+      var data = cache.within(start, end);
+      if (data.length < 10) {
+        do_query(start, end, function _iter_ (resp) {
+          days[Date.create(day).format(DATE_FMT)] = true;
+          collate(resp);
+        });
+      } else {
+        days[Date.create(day).format(DATE_FMT)] = true;
+        payload = payload.concat(data);
+        if (Object.keys(days).length >= range.length) {
+          do_payload( );
+        }
+      }
+    });
+
+  }
+
+  function accrue (begin, end, cb) {
+    var q = {
+      start: Date.create(Date.create(start).format(DATE_FMT))
+    , end: Date.create(Date.create(end).format(DATE_FMT))
+    };
+    var range = d3.time.days(q.start, q.end);
+    if (range.length > 3) {
+      var half = range.length / 2;
+      var head = range.slice(0, half);
+      var tail = range.slice(half);
+      // head.reduce(iter, { });
+      // tail.reduce(iter, { });
+    }
+  }
+
+  function collate (resp) {
+    console.log('payload', payload.length, 'resp', resp.length);
+    payload = payload.concat(resp);
+    cache.add(resp);
+    payload.sort(cmp_dateString);
+    var range = d3.time.days(query.start, query.end);
+    if (Object.keys(days).length >= range.length) {
+      do_payload( );
+    }
+  }
+
+  function cmp_dateString (a, b) {
+    return a.dateString > b.dateString;
+  }
+
+  function do_query (begin, end, cb) {
+    var range = d3.time.days(Date.create(begin), Date.create(end));
+    var fetch = url + param_string(begin, end);
+    console.log('QUERY FOR', range.length, 'days', query, fetch);
+    $.getJSON(fetch, cb);
+  }
+  
+  function first_page (resp) {
+    // console.log('resp', resp);
+    payload = resp;
+    do_payload( );
+  }
+
+  function do_payload ( ) {
+    if (opts.callback && opts.callback.call) {
+      opts.callback(payload);
+    } else {
+      // console.log(payload);
+    }
+  }
+
+  function init ( ) {
+    if (query.begin && query.end) {
+      refresh(query.begin, query.end);
+      // do_query(query.begin, query.end, first_page);
+    }
+    page( );
+    return page;
+  }
+
+  page.refresh = refresh;
+
+  return init( );
+}
+
+function time_in_range (data, opts) {
+  var range = {
+    high: 180
+  , low: 80
+  };
+
+  function sgv (d) { return d.sgv; }
+  function is_clean (d) { return sgv(d) > 39; }
+  function is_in_range (d) { return sgv(d) >= range.low && sgv(d) <= range.high; }
+  function is_high (d) { return sgv(d) > range.high; }
+  function is_low (d) { return sgv(d) < range.low; }
+  var results = [ ];
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["results" is defined but never used](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2715308466/issues/source?bid=2452374&fileBranchId=3345581#l416)